code/game/objects/machinery/miner.dm

```diff
@@ -0,0 +1,197 @@
+		"<span class='notice'>You fumble around figuring out [src]'s tubing and plating.</span>")
+		var/fumbling_time = 10 SECONDS - 2 SECONDS * user.skills.getRating("engineer")
+		if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+			return FALSE
+	playsound(loc, 'sound/items/ratchet.ogg', 25, TRUE)
+	user.visible_message("<span class='notice'>[user] starts repairing [src]'s tubing and plating.</span>",
+	"<span class='notice'>You start repairing [src]'s tubing and plating.</span>")
+	if(!do_after(user, 150, TRUE, src, BUSY_ICON_BUILD))
+		return FALSE
+	if(miner_status != MINER_SMALL_DAMAGE)
+		return FALSE
+	playsound(loc, 'sound/items/ratchet.ogg', 25, TRUE)
+	obj_integrity = max_integrity
+	set_miner_status()
+	user.visible_message("<span class='notice'>[user] repairs [src]'s tubing and plating.</span>",
+	"<span class='notice'>You repair [src]'s tubing and plating.</span>")
+	start_processing()
+	return TRUE
+
+/obj/machinery/miner/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	switch(miner_status)
+		if(MINER_DESTROYED)
+			to_chat(user, "<span class='info'>It's heavily damaged, and you can see internal workings.</span>\n<span class='info'>Use a blowtorch, then wirecutters, then a wrench to repair it.</span>")
+		if(MINER_MEDIUM_DAMAGE)
+			to_chat(user, "<span class='info'>It's damaged, and there are broken wires hanging out.</span>\n<span class='info'>Use wirecutters, then wrench to repair it.</span>")
+		if(MINER_SMALL_DAMAGE)
+			to_chat(user, "<span class='info'>It's lightly damaged, and you can see some dents and loose piping.</span>\n<span class='info'>Use a wrench to repair it.</span>")
+		if(MINER_RUNNING)
+			to_chat(user, "<span class='info'>[src]'s internal storage currently has [stored_mineral] sheets stored.</span>")
+
+/obj/machinery/miner/attack_hand(mob/living/user)
+	if(miner_status != MINER_RUNNING)
+		to_chat(user, "<span class='warning'>[src] is damaged!</span>")
+
+		return
+	if(!stored_mineral)
+
+		to_chat(user, "<span class='warning'>[src]'s internal storage currently has no minerals stored!</span>")
+
+		return
+	new mineral_produced(user.loc, stored_mineral)
+	stored_mineral = 0
+	start_processing()
+
+/obj/machinery/miner/process()
+	if(miner_status != MINER_RUNNING)
+		stop_processing()
+		return
+	if(add_tick >= required_ticks)
+		stored_mineral += 1			
+		add_tick = 0
+	if(stored_mineral >= 50)
+		stop_processing()
+	else
+		add_tick += 1
+
+/obj/machinery/miner/attack_alien(mob/living/carbon/xenomorph/xeno_attacker)
+	xeno_attacker.do_attack_animation(src, ATTACK_EFFECT_CLAW)
+	xeno_attacker.visible_message("<span class='danger'>[xeno_attacker] slashes \the [src]!</span>", \
+	"<span class='danger'>We slash \the [src]!</span>", null, 5)
+	playsound(loc, "alien_claw_metal", 25, TRUE)
+	if(miner_status == MINER_DESTROYED)
+		to_chat(xeno_attacker, "<span class='warning'>[src] is already destroyed!</span>")
+		return
+	obj_integrity -= 25

```

Comment:
Never change `obj_integrity` directly. Use the `take_damage` proc instead.

Reply:
in this case its okay to do it otherwise it will cause a ton of issues that arent easily solved.

Reply:
What kind of issues?

Reply:
mostly stuff related to obj_break() and deconstruction()
also see the base obj/attack_alien() proc

ideally someone would make obj/attack_alien() and overrides better and less code duped but outside the scope.

Reply:
`/obj/attack_alien()` is calling `/obj/proc/attack_generic()` which calls for `take_damage()` correctly.

I see the `INDESTRUCTIBLE` resistance flag doesn't allow the usage of `obj_integrity` through `take_damage()`. In this case it doesn't fit the machine, it's not indestructible, it just doesn't get deleted.

The ideal solution would be to make a distinction between unbreakable and indestructible things, which are both merged into the same flag right now. But that's a bigger refactor.

My suggestion is to remove the `INDESTRUCTIBLE` flag and redefine `obj_destruction()` instead. That or using a different variable than `obj_integrity`. Neither are really ideal.